### PR TITLE
send empty SAML service provider preset values as `unspecified` to Web UI.

### PR DESCRIPTION
**Pull request by flyinghermit**
This is added to unified resource response so that web UI does not need to think about empty string to `unspecified` conversion.

### CODE REVIEW

File: lib/web/ui/app.go [hunk 1]
```diff
@@ -155,8 +155,14 @@ func MakeApp(app types.Application, c MakeAppsConfig) App {
...
 // MakeAppTypeFromSAMLApp creates App type from SAMLIdPServiceProvider type for the WebUI.
 // Keep in sync with lib/teleterm/apiserver/handler/handler_apps.go.
+// Note: The SAMLAppPreset field is used in SAML service provider update flow in the
+// Web UI. Thus, this field is currently not available in the Connect App type.
 func MakeAppTypeFromSAMLApp(app types.SAMLIdPServiceProvider, c MakeAppsConfig) App {
 	labels := makeLabels(app.GetAllLabels())
+	samlAppPreset := "unspecified"
+	if app.GetPreset() != "" {
+		samlAppPreset = app.GetPreset()
+	}
```
Comment (zmb3) [inline lib/web/ui/app.go:165]:
```suggestion
	samlAppPreset := cmp.Or(app.GetPreset(), "unspecified")
```

Comment (flyinghermit) [reply lib/web/ui/app.go]: Wow, didn't knew this was possible. I never used the cmp package besides the Diff func in tests.

https://github.com/gravitational/teleport/pull/43863/commits/b60eb520684adb661c850f47d21691a22739e3c3

Comment (zmb3) [reply lib/web/ui/app.go]: That's go-cmp, not cmp from the standard library.

**Review by zmb3 — APPROVED** [review]

**Comment by flyinghermit** [comment]
Friendly ping @strideynet @camscale

**Review by camscale — APPROVED** [review]

**Pull request merged**